### PR TITLE
feat(source): add managed source overview surfaces

### LINKED ISSUE #372: feat(source): add managed source overview to daemon status and source list
## Summary

Managed sources currently have a detailed per-source status command:

- `uxc source status <namespace> <source_key>`

But they do not have a good top-level overview surface.

Today:
- `uxc daemon status` only shows daemon-wide MCP/session summary fields
- `uxc daemon sessions` shows session-level details
- `uxc source status` requires the caller to already know `<namespace>` and `<source_key>`

That makes managed sources harder to discover and inspect operationally.

### 1. Add managed source summary fields to `uxc daemon status`

Keep this at the summary/aggregate level, not full per-source detail.

Suggested fields:
- `managed_sources`
- `managed_sources_running`
- `managed_streams`

Optional future field:
- `managed_source_events_written`

The goal is to let users quickly see whether the daemon is managing sources at all, without turning `daemon status` into a mixed detail dump.

### 2. Add `uxc source list`

Add a top-level listing command so users can discover managed sources without already knowing the key.

Suggested shape for each entry:
- `namespace`
- `source_key`
- `status`
- `run_id`
- `stream_id`
- `updated_at_unix`
- `last_error` (optional)

This should become the primary overview surface for managed sources, while:
- `uxc source status <namespace> <source_key>` stays the detailed single-source view
- `uxc daemon status` stays the daemon-wide summary

## Why

This keeps the product surface layered and easier to understand:
- `uxc daemon status`: daemon health and aggregate counts
- `uxc daemon sessions`: session-level diagnostics
- `uxc source list`: managed source overview
- `uxc source status`: managed source detail

Without `source list`, users have no convenient way to discover existing managed sources, and without summary fields in `daemon status`, managed sources are mostly invisible at the daemon overview level.

### PR DESCRIPTION
**Pull request by jolestar**
## What
Add managed source overview surfaces to the daemon and client APIs.

## Why
Managed sources already have per-source detail via `uxc source status`, but there is no good top-level way to discover them or see whether the daemon is managing any at all.

Closes #372

## How
Added aggregate managed-source counts to `uxc daemon status`, introduced a new `source.list` daemon RPC plus `uxc source list`, and exposed the new data through both the Rust and TypeScript daemon clients.

## Changes
- [x] Add managed source summary fields to `uxc daemon status`
- [x] Add `source.list` RPC and `uxc source list`
- [x] Expose source list and new daemon status fields in Rust/TS daemon clients
- [x] Add CLI, daemon, and TS client test coverage

### Unit Tests
- [x] New tests added where needed
- [ ] Full `cargo test` suite run

### Manual Testing
- [ ] Tested against real APIs
- [x] Tested edge cases relevant to daemon status fallback and source discovery
- [ ] Tested error conditions beyond automated coverage

### Test Results
```bash
cargo check -q
cargo test --test daemon_cli_test daemon_start_status_stop_lifecycle -- --test-threads=1
cargo test managed_source_list_and_daemon_status_expose_overview_counts --lib -- --test-threads=1
cargo test --test cli_help_regression_test -- --test-threads=1
npm test --workspace @holon-run/uxc-daemon-client -- client.test.ts
```

## Checklist

- [x] Code formatted with `cargo fmt`
- [ ] No clippy warnings (`cargo clippy -- -D warnings`)
- [ ] All tests pass (`cargo test`)
- [x] Documentation updated via CLI/help surfaces
- [x] Commit messages follow convention
- [x] PR title follows convention

## Breaking Changes
None.

## Additional Notes
`daemon status` keeps managed-source data at the aggregate level, while `source list` is the new overview surface and `source status` remains the per-source detail surface.

### CODE REVIEW

File: src/daemon.rs [hunk 1]
```diff
@@ -2897,6 +2911,20 @@ impl ManagedSourceManager {
         Ok(view_from_record(&record))
     }
 
+    async fn list(&self) -> Result<Vec<ManagedSourceListEntry>> {
+        let records = self.store.load_sources().await?;
+        Ok(records.iter().map(list_entry_from_record).collect())
```
[the remark this answers is not in the document]
Comment (jolestar) [reply src/daemon.rs]: Fixed in 2981c12. `source.list` now uses a dedicated store query that selects only the overview columns instead of loading full managed-source rows.

File: src/daemon.rs [hunk 2]
```diff
@@ -3839,6 +3879,8 @@ impl DaemonRuntime {
     pub async fn status(&self) -> DaemonStatus {
         let state = self.state.lock().await;
         let (stdio_sessions, http_sessions, reuse_hits) = self.mcp.status_counts().await;
+        let (managed_sources, managed_sources_running, managed_streams) =
+            self.managed_sources.summary().await.unwrap_or((0, 0, 0));
```
[the remark this answers is not in the document]
Comment (jolestar) [reply src/daemon.rs]: Fixed in 2981c12. `DaemonRuntime::status()` now logs a warning before falling back to zero managed-source counts when the store summary read fails.
